.github/workflows/linux_builds.yml

```diff
@@ -26,62 +25,89 @@ jobs:
-            bin: "./bin/godot.linuxbsd.editor.dev.double.x86_64.san"
+            bin: ./bin/godot.linuxbsd.editor.dev.double.x86_64.san
             build-mono: false
             tests: true
             proj-test: true
-            # Generate an API dump for godot-cpp tests.
-            api-dump: true
             # Skip 2GiB artifact speeding up action.
             artifact: false
 
           - name: Editor with clang sanitizers (target=editor, tests=yes, dev_build=yes, use_asan=yes, use_ubsan=yes, use_llvm=yes, linker=lld)
             cache-name: linux-editor-llvm-sanitizers
             target: editor
             sconsflags: dev_build=yes use_asan=yes use_ubsan=yes use_llvm=yes linker=lld
-            bin: "./bin/godot.linuxbsd.editor.dev.x86_64.llvm.san"
+            bin: ./bin/godot.linuxbsd.editor.dev.x86_64.llvm.san
             build-mono: false
             tests: true
             # Skip 2GiB artifact speeding up action.
             artifact: false
+            # Test our oldest supported SCons/Python versions on one arbitrary editor build.
+            legacy-scons: true
+
+          - name: Editor with ThreadSanitizer (target=editor, tests=yes, dev_build=yes, use_tsan=yes, use_llvm=yes, linker=lld)
+            cache-name: linux-editor-thread-sanitizer
+            target: editor
+            tests: true
+            sconsflags: dev_build=yes use_tsan=yes use_llvm=yes linker=lld
+            bin: ./bin/godot.linuxbsd.editor.dev.x86_64.llvm.san
+            build-mono: false
+            # Skip 2GiB artifact speeding up action.
+            artifact: false
 
-          - name: Template w/ Mono (target=template_release)
+          - name: Template w/ Mono, release (target=template_release, tests=yes)
             cache-name: linux-template-mono
             target: template_release
             sconsflags: module_mono_enabled=yes
+            bin: ./bin/godot.linuxbsd.template_release.x86_64.mono
             build-mono: false
-            tests: false
+            tests: true
             artifact: true
 
-          - name: Minimal template (target=template_release, everything disabled)
+          - name: Template w/ Mono, debug (target=template_debug, tests=yes)
+            cache-name: linux-template-mono-debug
+            target: template_debug
+            sconsflags: module_mono_enabled=yes
+            bin: ./bin/godot.linuxbsd.template_debug.x86_64.mono
+            build-mono: false
+            tests: true
+            artifact: true
+
+          - name: Minimal template (target=template_release, tests=yes, everything disabled)
             cache-name: linux-template-minimal
             target: template_release
             sconsflags: modules_enabled_by_default=no disable_3d=yes disable_advanced_gui=yes deprecated=no minizip=no
-            tests: false
+            bin: ./bin/godot.linuxbsd.template_release.x86_64
+            tests: false  # Needs various fixes to work on minimal builds.
```

Comment:
Unlike the `4.2` PR, I didn't enable tests here for the minimal template build, as multiple PRs would need to be backported for it, and that's outside the scope of what I wanted to do with this CI sync.

Reply:
Actually I'm skipping tests for all template builds, more PRs would need to be backported.

I don't think it's worth the trouble for an old branch like `4.1`, but if someone wants to it should be doable in a follow-up PR.